addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/model/Realtime.java

```diff
@@ -18,31 +16,51 @@
  */
 public class Realtime extends ErrorResponse {
 
-    @SerializedName(value = "current", alternate = "current_ma")
+    private static final int V2_TO_V1 = 1000;
+
     private double current;
-    @SerializedName(value = "power", alternate = "power_mw")
     private double power;
-    @SerializedName(value = "total", alternate = "total_wh")
     private double total;
-    @SerializedName(value = "voltage", alternate = "voltage_mv")
     private double voltage;
 
+    // JSON names used for v2 hardware
+    private double current_ma;
+    private double power_mw;
+    private double total_wh;
+    private double voltage_mv;
+
     public double getCurrent() {
+        if (current == 0 && current_ma != 0) {
+         return current_ma / V2_TO_V1; // v1 displays as amps
+        }
+
         return current;
     }
 
     public double getPower() {
+        if (power == 0.0 && power_mw != 0.0) {
```

Comment:
Just simplify to `return powerMw > 0.0 ? powerMw / MILLIWATT_TO_WATT : power;` Because the constant explains the coversion no need to add comments. Also it's preferable not to compare doubles to exact values, so therefor the `>`. When powerMw has value we can safely assume power itself is not set so no need to test that value.
Also use this pattern for the other 3 methods.